streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNode.java

```diff
@@ -29,14 +30,20 @@
     private ProcessorContext context;
     private Deserializer<K> keyDeserializer;
     private Deserializer<V> valDeserializer;
+    private final TimestampExtractor timestampExtractor;
 
-    public SourceNode(String name, List<String> topics, Deserializer<K> keyDeserializer, Deserializer<V> valDeserializer) {
+    public SourceNode(String name, List<String> topics, TimestampExtractor timestampExtractor, Deserializer<K> keyDeserializer, Deserializer<V> valDeserializer) {
```

Comment:
@jeyhunkarimov Sorry for jumping in a little late.  @guozhangwang  is correct about the test.

An example for unit-testing with regex defined topics using the `TopologyBuilder` is `TopologyBuilderTest#shouldSetCorrectSourceNodesWithRegexUpdatedTopics` (line 675)